src/Notifier.cpp

```diff
@@ -61,6 +63,24 @@ bool Notifier::Initialize(const size_t handshakeCount) {
 }
 
 /*
+ * Receives message data without caring about handshakes, added sleeping code
+ * to prevent pounding the cpu
+ */
+std::string Notifier::ReceiveData() {
+   StopWatch waitCheck;
+   while (waitCheck.ElapsedSec() < mMaxTimeoutInSec && mKeepRunning.load()) {
```

Comment:
Or you create a function that checks this. In your unit tests you then mock it out so that you can trigger an early exit.